stelaval/assignment4/IslandsCounter.java

```diff
@@ -0,0 +1,57 @@
+package assignment4;
+
+public class IslandsCounter {
+  
+  /**
+   * Counts the number of islands in a 2-dimensional map of tiles.
+   * Each tile is either land or water.
+   * Two tiles belong to the same island if they are both land and are adjacent
+   * horizontally or vertically, but not diagonally.
+   * 
+   * @param map is a 2-dimensional array of booleans,
+   * where false means water and true means land
+   * 
+   * @return the number of islands
+   */
+  public int countIslands(boolean[][] map) {
+    if (map.length == 0 || map[0].length == 0) {
+      throw new IllegalArgumentException("Map of tiles is empty.");
+    }
+    int rows = map.length;
+    int columns = map[0].length;
+    boolean[][] visited = new boolean[rows][columns];
+
+    int count = 0;
+    for (int i = 0; i < rows; i++) {
+      for (int j = 0; j < columns; j++) {
+        if (map[i][j] == true && !visited[i][j]) {

```

Comment:
The `== true` part is not really necessary. Feel free to leave it though if you find it more readable.